### PR TITLE
feat: simpler glob matching

### LINKED ISSUE #904: Global metadata: simplify path matching

### PR DESCRIPTION
**Pull request by rofe**
Fix #904

### CODE REVIEW

File: src/extract-metadata.js [hunk 1]
```diff
@@ -120,13 +119,20 @@ function applyMetaRule(target, obj) {
   });
 }
 
+function globToRegExp(glob) {
+  let reString = glob.replace(/\*\*/g, '_');
+  reString = reString.replace(/\*/g, '[0-9a-z-]*');
+  reString = reString.replace(/_/g, '.*');
```
Comment (tripodsan) [inline src/extract-metadata.js:125]:
```suggestion
  let reString = glob
    .replace(/\*\*/g, '_')
    .replace(/\*/g, '[0-9a-z-]*')
    .replace(/_/g, '.*');
```

File: src/extract-metadata.js [hunk 2]
```diff
@@ -120,13 +119,20 @@ function applyMetaRule(target, obj) {
...
+  return new RegExp(`^${reString}`);
+}
+
 function filterGlobalMetadata(metaRules, url) {
   const metaConfig = {};
   metaRules.forEach((rule) => {
     const glob = rule.url || rule.URL || rule.Url;
-    if (glob && typeof glob === 'string') {
+    if (glob && typeof glob === 'string' && /[0-9a-z-/*]/.test(glob)) {
```
Comment (tripodsan) [inline src/extract-metadata.js:133]: are you sure about this additional test?

**Review by tripodsan — APPROVED** [review]

File: src/extract-metadata.js (hunk 2 above)
Comment (rofe) [reply src/extract-metadata.js:134]: Yes, we want to make sure nothing funny will bleed through to the regexp.

**Comment by rofe** [comment]
`run_diff_tests` is expected to fail due to different strains being used (`breaking-202103` vs `breaking-800`)

**Pull request merged**